tests/test_reserveSourcesTask.py

```diff
@@ -0,0 +1,154 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import unittest
+import numpy as np
+
+import lsst.afw.table
+import lsst.meas.algorithms
+import lsst.utils.tests
+
+from lsst.pipe.base import Struct
+
+
+class ReserveSourcesTaskTest(lsst.utils.tests.TestCase):
+    """TestCase for the ReserveSourcesTask"""
+    def setUp(self):
+        self.num = 100  # Number of sources
+
+    def construct(self, name, fraction):
+        """Construct the test environment
+
+        This isn't called 'setUp' because we want to vary the `fraction`.
+
+        Parameters
+        ----------
+        name : `str`
+            Name of column for flagging reserved sources (without "_reserved").
+        fraction : `float`
+            Fraction of sources to reserve.
+
+        Return struct elements
+        ----------------------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Catalog of sources.
+        task : `lsst.meas.algorithms.ReserveSourcesTask`
+            Task to do the reservations.
+        key : `lsst.afw.table.Key`
+            Key to the flag column.
+        """
+        schema = lsst.afw.table.SourceTable.makeMinimalSchema()
+        config = lsst.meas.algorithms.ReserveSourcesConfig()
+        config.fraction = fraction
+        task = lsst.meas.algorithms.ReserveSourcesTask(columnName=name, schema=schema,
+                                                       doc="Documentation is good", config=config)
+        key = schema[name + "_reserved"].asKey()
+        catalog = lsst.afw.table.SourceCatalog(schema)
+        catalog.reserve(self.num)
+        for _ in range(self.num):
+            catalog.addNew()
+        return Struct(catalog=catalog, task=task, key=key)
+
+    def check(self, sources, task, key, fraction):
+        """Check that source reservation is working
+
+        We test that the source reservation works, that it depends
+        on the RNG seed and that thigns behave as we expect when there's
+        a prior selection.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.Catalog` or `list` of `lsst.afw.table.Record`
+            List of sources.
+        task : `lsst.meas.algorithms.ReserveSourcesTask`
+            Task to do the reservations.
+        key : `lsst.afw.table.Key`
+            Key to the flag column.
+        fraction : `float`
+            Fraction of sources to reserve.
+        """
+        numExpect = fraction*len(sources)
+
+        # No prior
+        results1 = task.run(sources, expId=1)
+        flagged1 = np.array([ss.get(key) for ss in sources])
+        self.assertEqual(flagged1.sum(), numExpect)
```

Comment:
For this and the other asserts, define a message like `"fraction=%s, key=%s"%(fraction,key)`, to simplify debugging (`longMessage=True` above will print this in addition to the usual message),

Reply:
Oh, if all I have to do is the above then that's easy.